Bottom: 8ba0543f8e33df7da7e1978a46c15eea3b57a40e
Top:    12213f626c9659bb5f273acf8ef6099248bcb468
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-11 15:31:16 +0000

Cure crosshair lagging on screen











---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index ec0018a..258904b 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -1117,6 +1117,7 @@ ghid_port_window_motion_cb (GtkWidget * widget,
   static gint x_prev = -1, y_prev = -1;
   gboolean moved;
   GdkModifierType state;
+#if 0
   GdkGLContext* pGlContext = gtk_widget_get_gl_context (widget);
   GdkGLDrawable* pGlDrawable = gtk_widget_get_gl_drawable (widget);
 
@@ -1125,6 +1126,7 @@ ghid_port_window_motion_cb (GtkWidget * widget,
     printf ("GL THingy returned\n");
     return FALSE;
   }
+#endif
 
   state = (GdkModifierType) (ev->state);
   mk = ghid_modifier_keys_state (&state);
@@ -1132,8 +1134,11 @@ ghid_port_window_motion_cb (GtkWidget * widget,
   if ((ev->state & GDK_BUTTON3_MASK) == GDK_BUTTON3_MASK
       && mk == NONE_PRESSED)
     {
+/* Not sure this really matters, since we're using invalidate to get a redraw */
+#if 1
       if (gtk_events_pending ())
 	return FALSE;
+#endif
       dx = gport->zoom * (x_prev - ev->x);
       dy = gport->zoom * (y_prev - ev->y);
       if (x_prev > 0)
@@ -1145,10 +1150,11 @@ ghid_port_window_motion_cb (GtkWidget * widget,
   x_prev = y_prev = -1;
   moved = ghid_note_event_location (ev);
 
-  ghid_show_crosshair (TRUE);
-  if (moved && have_crosshair_attachments ())
+//  ghid_show_crosshair (TRUE);
+  if (moved) // && have_crosshair_attachments ())
     ghid_draw_area_update (gport, NULL);
 
+#if 0
   if (gdk_gl_drawable_is_double_buffered (pGlDrawable))
     gdk_gl_drawable_swap_buffers (pGlDrawable);
   else
@@ -1156,6 +1162,8 @@ ghid_port_window_motion_cb (GtkWidget * widget,
 
   /* end drawing to current GL-context */
   gdk_gl_drawable_gl_end (pGlDrawable);
+#endif
+
   return FALSE;
 }
